pez-globo/pufferfish-vent-software
# Refactor public interfaces of some low-level driver code for the Nonin sensor

This PR includes:
- remove enums that were not used
- Reorder enums, consts to Types.h
- Rename `SignalAmplitude` to `SignalQuality`
- Rename `Pleth` to `PlethPulseAmplitudes`
- Rename `PacketMeasurements` to `Sample`
- Rename `StatusByteStruct` to `Status`
- Reorder `StatusByteStruct` fields as per datasheet
- Reorder fields in Sample as per datasheet (pg 6, Serial Data format (2) table)
- Rename `heart_rate` to `hr`
- Rename `spo2_d_beat` to `spo2_b_b`
- Rename `spo2_d_fast` to `spo2_fast` (as per datasheet)
- Rename `nonin_oem_revision` to `firmware_revision`
- Remove unnecessary code comments
- Use FrameBufferStatus for FrameBuffer instead of BufferStatus
- Rename `get_6bit_data` to `get_spo2_data` as it essentialy only gets SpO2 data from the packet
- Rename `get_9bit_data` to `get_hr_data`
- Rename `mask_6bit` to `spo2_missing_data` which is equal to `0x7F` or 127, this is an indicator for missing SpO2 data, we just ignore it if we get these bytes
- Rename `mask_9bit` to `hr_missing_data` which is `0x01FF` or 511 this is an indicator for missing HR data, we just ignore it if we get these bytes
- Rename `FrameInputStatus` `available` enum to `output_ready`
- Rename `FrameInputStatus` `framing_error` enum to `checksum_failed`
- Rename `PacketStatus` `framing_error` enum to `checksum_failed`
- Rename `PacketStatus` `missed_data` enum to `frame_loss`
- Rename `PacketInputStatus` `missed_data` enum to `frame_loss`
- Rename `PacketStatus` `available` enum to `ok`

TODO: check for other changes
